common/txn.ts

```diff
@@ -1,34 +1,31 @@
 // A txn (pronounced "texan") respresents a payment between two ids on Manifold
 // Shortened from "transaction" to distinguish from Firebase transactions (and save chars)
-export type Txn = {
+export type AnyTxnType = Donation | Tip
+
+export type Txn<T extends AnyTxnType = AnyTxnType> = {
   id: string
   createdTime: number
 
   fromId: string
-  fromType: SourceType
-
   toId: string
-  toType: SourceType
```

Comment:
Are we ever going to have txns without `fromType` or without `toType?` I think it should be required in every Txn. I suggest we don't delete these lines. Even with these lines, the types should still get narrowed correctly.

<img width="413" alt="image" src="https://user-images.githubusercontent.com/18368938/174910130-521daa77-4c10-4337-bf15-1b61177bdf1c.png">
(In the above example, the last two lines complain that "BANK" is not assignable to "USER" and that c.z has no type overlap with "BANK")

Reply:
An alternative approach: instead of `Txn<T extends AnyTxnType>` do `Txn<T extends { fromId: SourceType,  toId...`

Reply:
I just deleted these declarations because they are redundant, but the facts that you are displeased and that Austin instantly misunderstood it seems like good evidence that it's good documentation, so I will put them back.

I guess nobody noticed when I analogously removed the redundant `outcomeType` declaration from `Contract`.